### ISSUE TITLE
Node creation redirects (via `$form_state['redirect']`) do not work.

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#6424]
I am attempting to redirect people after they create a node, so rather than landing on `node/123` they get taken to a custom layout I have built. Unfortunately, the usual trick of using `$form_state['redirect']` to allow the form processing to complete before redirection, does not appear to work for the node form. 

Instead, I've had to stick a `backdrop_goto()` into `hook_node_insert()` which prevents the message from being set (and potentially other important things from happening!)

Below is a custom module that should work (I think?), but it does not.  Note: this does NOT work on Drupal 7 either. 

```php
/**
 * Implements hook_form_FORM_ID_alter().
 */
function customtest_form_node_form_alter(&$form, &$form_state) {
	// Add our own submit handler.
	$form['#submit'][] = 'customtest_node_form_submit';
}

/**
 * Custom submit handler. @see customtest_form_node_form_alter().
 */
function customtest_node_form_submit(&$form, &$form_state) {
	// Not working?!
	$form_state['redirect'] = '/';
}
```

If someone else knows another way to add a redirect when a node is created (no cheating using the rules module!) I'd love to learn :)

**Comment by argiepiano** [comment]
Yeah, this is one of those inconsistencies with node submissions. The problem is that you are adding the additional submit handler to the **form** itself rather than to the actual submit element. This typically is not a problem with other forms, but it is a problem with node forms, basically because `node_form_submit()` invokes the form submit handlers **before** settings its own `$form_state['redirect'] = 'node/' . $node->nid . '/edit';`  (those submit handlers are invoked manually by `node_form_submit_build_node()` which is called by `node_form_submit()`).

Sooo... the solution is to add the custom callback as follows:

```php
  $form['actions']['submit']['#submit'][] = 'customtest_node_form_submit';
```

Try that and see if that works.
